tools/asset-sync/assets.Tests.Helpers.ps1

```diff
@@ -0,0 +1,208 @@
+# This powershell script contains multiple functions that are dot-included in the BeforeAll function of assets.Tests.ps1
+#
+# Describe-TestFolder is the primary entrypoint, and is used to create a test environment given some basic setup input. This function is used to:
+#  - Create faux recording.jsons and place them in a fake language repo of any structure
+#  - Create any fake language repo structure with some random sample data
+#  - Dynamically create a copy of the assets repo within a local folder to actually create real test scenarios
+#  - Will push duplicates of test scenario branches to alternative, guid-fied branch names for actual push/pull operations.
+
+Function Get-Basic-AssetsJson {
+  param(
+    [boolean]$RandomizeRepoId = $true
+  )
+  $result = (Get-Content -Raw -Path (Join-Path $PSScriptRoot "assets.json") | ConvertFrom-Json)
+
+  if ($RandomizeRepoId) {
+    Add-Member -InputObject $result -MemberType "NoteProperty" -Name "AssetRepoId" -Value (New-Guid).ToString()
+  }
+
+  return $result
+}
+
+Function Get-Full-TestPath {
+  Param(
+      [string] $Path
+  )
+  return $Path.Replace('TestDrive:', (Get-PSDrive TestDrive).Root)
+}
+
+Function Get-TestFolder {
+  return (Join-Path $PSScriptRoot ".testruns")
+}
+
+Function Get-TestPath {
+  $disablePermanentStorage = $env:USE_TESTDRIVE
+
+  $localTempPath = Get-TestFolder
+
+  if ($disablePermanentStorage){
+    $testPath = "TestDrive:\$([Guid]::NewGuid())\"
+    
+    mkdir $testPath | Out-Null
+    return $testPath
+  }
+  else {
+    $testPath = (Join-Path $localTempPath "$([Guid]::NewGuid())")
+
+    mkdir -p $testPath | Out-Null
+    return $testPath
+  }
+}
+
+Function Initialize-Integration-Branches {
+  param(
+    [PSCustomObject] $Config,
+    [string] $TestGuid
+  )
+
+  $adjustedBranchName = "test_$($TestGuid)_$($Config.AssetsRepoBranch)"
+  $tempPath = "TestDrive:\$([Guid]::NewGuid())\"
+  mkdir $tempPath | Out-Null
+  
+  try {
+    Push-Location $tempPath
+    Write-Host "git clone https://github.com/Azure/azure-sdk-assets-integration ."
+    git clone https://github.com/Azure/azure-sdk-assets-integration .
```

Comment:
This won't work for everybody. For example I have SSH credentials set up for `git@github.com:` type URLs. I'm not sure what the best solution is here but I'm sure you're not the first to encounter this problem.

Reply:
I'll abstract this to a `Resolve-AssetsRepoRemote` so that we can handle those different cases in a function (and unit test them).